core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -355,6 +377,43 @@ write_files:
     content: {{$w.GzippedBase64Content}}
   {{- end }}
 {{- end }}
+{{if and (.AmazonSsmAgent.Enabled) (ne .AmazonSsmAgent.DownloadUrl "")}}
+  - path: "/opt/bin/ssm/install-ssm-agent.sh"
+    permissions: 0700
+    content: |
+      #!/bin/bash
+      set -eo pipefail
+
+      function finish {
+        if [ "$(pwd)" == "/opt/bin/ssm" ]; then
+          popd > /dev/null 2>&1
+        fi
+      }
+
+      trap finish EXIT
+
+      pushd /opt/bin/ssm > /dev/null 2>&1
+
+      TAR_FILE=ssm.linux-amd64.tar.gz
+      CHECKSUM_FILE="${TAR_FILE}.sha1"
+
+      curl --silent -L -o ${TAR_FILE} "{{ .AmazonSsmAgent.DownloadUrl }}"
+      if [[ -f "${CHECKSUM_FILE}" ]]; then
+        ( sha1sum --status -c "${CHECKSUM_FILE}" && exit 0 ) || true
```

Comment:
Just a note of warning: Changing node config using agent is a slippery slope, where whole set of nodes can be managed out of bounds for the sake of simplicity only to find out that ASG might restart nodes and new ones come up with unaltered “old” config. It is easy to take it further and turn your cluster into a snowflake, where ‘ kube-aws update’ is not enough to bring up k8s cluster as some set of changes need to be applied which are not recorded in SCM where you do kube-aws update from.
 
 
Now back to your script. Just try updating URL and try running it yourself. ‘( .... && exit 0 )’ isn't doing what you think it is doing